cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/GroupedMessageIteratorProvider.java

```diff
@@ -94,12 +104,78 @@ private CassandraGroupedMessageFilter createInitialFilter(GroupedMessageFilter f
 				book.getId().getName(),
 				getFirstPage().getId().getName(),
 				filter.getGroupName(),
-				filter.getFrom(),
-				filter.getTo(),
+				leftBoundFilter,
+				rightBoundFilter,
 				order,
 				filter.getLimit());
 	}
 
+	private FilterForGreater<Instant> createLeftBoundFilter(GroupedMessageFilter filter) throws CradleStorageException
+	{
+		FilterForGreater<Instant> result = filter.getFrom();
+		firstPage = FilterUtils.findFirstPage(filter.getPageId(), result, book);
```

Comment:
I don't really think that it is a good idea to set the field inside a method that says it `creates left boundary filter`. Probably, we should first identify the boundary and then set the `firstPage` property in the constructor.